app/services/user_signup.rb

```diff
@@ -0,0 +1,49 @@
+class UserSignup
+  attr_reader :status, :error_message
+
+  def initialize(user)
+    @user = user
+  end
+
+  def sign_up(options = {})
+    stripe_token = options[:stripe_token]
+    invitation_token = options[:invitation_token]
+    if @user.valid?
+      charge = StripeWrapper::Charge.create(token: stripe_token, customer: @user.name, amount: 999, description: "MyFLiX Signup fee for #{@user.email}")
+
+      if charge.succesful?
+        @user.stripe_customer_id = charge.customer_id
+        @user.save
+        handle_invitation(invitation_token)
+
+        UserMailer.delay.welcome(@user.id)
+        @status = :success
+        self
+      else
+        @status = :failed
+        @error_message = charge.error_message
+        self
+      end
+    else
+      @status = :failed
+      @error_message = "There was an error creating your account. Please check the errors below."
+      self
```

Comment:
Haha I just catched this when refactoring the UserSignup for the changes in the StripeWrapper. I noticed exactly the same :+1: